packages/smooth_app/lib/pages/product/new_product_page.dart

```diff
@@ -0,0 +1,271 @@
+import 'dart:ui';
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:openfoodfacts/model/Attribute.dart';
+import 'package:openfoodfacts/model/Product.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:provider/provider.dart';
+import 'package:smooth_app/cards/data_cards/image_upload_card.dart';
+import 'package:smooth_app/cards/data_cards/score_attribute_card.dart';
+import 'package:smooth_app/cards/expandables/attribute_list_expandable.dart';
+import 'package:smooth_app/data_models/product_preferences.dart';
+import 'package:smooth_app/database/dao_product_extra.dart';
+import 'package:smooth_app/database/local_database.dart';
+import 'package:smooth_app/helpers/attributes_card_helper.dart';
+import 'package:smooth_app/helpers/launch_url_helper.dart';
+import 'package:smooth_app/pages/product/common/product_dialog_helper.dart';
+import 'package:smooth_app/themes/smooth_theme.dart';
+import 'package:smooth_ui_library/widgets/smooth_card.dart';
+
+class NewProductPage extends StatefulWidget {
+  const NewProductPage({
+    required this.product,
+  });
+
+  final Product product;
+
+  @override
+  State<NewProductPage> createState() => _ProductPageState();
+}
+
+class _ProductPageState extends State<NewProductPage> {
+  late Product _product;
+  bool _first = true;
+
+  @override
+  void initState() {
+    super.initState();
+    _updateHistory(context);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final LocalDatabase localDatabase = context.watch<LocalDatabase>();
+    final AppLocalizations appLocalizations = AppLocalizations.of(context)!;
+    if (_first) {
+      _first = false;
+      _product = widget.product;
+    }
+    return Scaffold(
+      backgroundColor: SmoothTheme.COLOR_PRODUCT_PAGE_BACKGROUND,
+      appBar: AppBar(
+        title: Text(_getProductName(appLocalizations)),
+        actions: <Widget>[
+          PopupMenuButton<String>(
+            itemBuilder: (final BuildContext context) =>
+                <PopupMenuEntry<String>>[
+              PopupMenuItem<String>(
+                value: 'web',
+                child: Text(appLocalizations.label_web),
+              ),
+              PopupMenuItem<String>(
+                value: 'refresh',
+                child: Text(appLocalizations.label_refresh),
+              ),
+            ],
+            onSelected: (final String value) async {
```

Comment:
nit: Should this handler be extracted in a method? It takes up 30 lines of code and has little to do with the layout.

Reply:
hmm, the thing is I'm not sure if this code will stay, but it's useful right now so I've copy pasted it as is from the old product_page. For now, created a method for refreshing the product page